Gesture and UIControl Extensions

barbarakos opened the pull request:
Gesture+Extensions
UIControl + Extensions

barbarakos committed:
Gesture and UIControl Extensions

Gesture+Extensions
UIControl + Extensions

barbarakos committed:
UIControl functions

Added UIControl+Extensions functions for textFieldDidEndEditing.

dorastipkovic commented on QuizApp/Leaderboard/ViewControllers/LeaderboardViewController.swift:
```diff
@@ -7,7 +7,7 @@ class LeaderboardViewController: UIViewController {
     private let insetToData = 50
 
-    private var cacellables = Set<AnyCancellable>()
+    private var cancellables = Set<AnyCancellable>()
```
Properties initialized in the same line as declared before those initialized at some other time
Check [style guide.](https://github.com/fiveagency/ios-swift-guide/tree/master/Code%20Style#property-declaration-order:~:text=properties%20initialized%20in%20the%20same%20line%20as%20declared%20before%20those%20initialized%20at%20some%20other%20time)

Check and apply on other places as well.

dorastipkovic commented on QuizApp/Login/ViewControllers/LoginViewController.swift:
```diff
@@ -28,7 +30,7 @@ class LoginViewController: UIViewController {
+    func handleLogIn() {
         guard
             let username = emailTextField.text,
             let password = passwordTextField.text
```
return in one line

dorastipkovic commented on QuizApp/User/ViewControllers/UserViewController.swift:
```diff
@@ -37,7 +37,7 @@ class UserViewController: UIViewController {
     }
 
-    @objc func handleLogOut() {
+    func handleLogOut() {
```
private

barbarakos committed:
Updated UIControl+Extensions

LoginViewController text fields subscribed to publishers in UIControl+Extensions.

dorastipkovic reviewed (changes requested):
Our usual practice is to have a `func bindViews()` in each ViewController and put all bindings there.

barbarakos committed:
PR changes

barbarakos committed:
Merge branch 'feature/leaderboard' into feature/GesturesExtension

# Conflicts:
#	QuizApp/Quiz/ViewControllers/QuizViewController.swift

dorastipkovic commented on QuizApp/Quiz/ViewControllers/QuizDetailsViewController.swift:
```diff
@@ -39,7 +41,16 @@ class QuizDetailsViewController: UIViewController {
+        leaderboardButton
+            .tap
+            .sink { _ in
+                self.showLeaderboard()
```
weak

dorastipkovic commented on QuizApp/Quiz/Views/QuizDetailsView.swift:
```diff
@@ -30,10 +34,19 @@ class QuizDetailsView: UIView {
+        startButton
+            .tap
+            .sink { _ in
+                self.startQuiz()
```
weak

dorastipkovic commented on QuizApp/Search/SearchViewController.swift:
```diff
@@ -56,6 +57,30 @@ class SearchViewController: UIViewController {
         }
     }
 
+    func textFieldDidBeginEditing(_ textField: UITextField) {
+        textField.layer.borderWidth = 1
+    }
+
+    func textFieldDidEndEditing(_ textField: UITextField) {
+        textField.layer.borderWidth = 0
+    }
+
+    func bindViews() {
+        searchBar.textField
+            .textDidBeginEditing
+            .sink { [weak self] _ in
+                self?.searchBar.textField.layer.borderWidth = 1
+            }
+            .store(in: &cancellables)
+
+        searchBar.textField
+            .textDidEndEditing
+            .sink { [weak self] _ in
+                self?.searchBar.textField.layer.borderWidth = 0
+            }
+            .store(in: &cancellables)
+    }
```
do you still need the `func textFieldDidBeginEditing(_ textField: UITextField)` and `func textFieldDidEndEditing(_ textField: UITextField)` now that you have the observers?

dorastipkovic commented on QuizApp/Search/SearchViewController.swift:
```diff
@@ -56,6 +57,30 @@ class SearchViewController: UIViewController {
+            }
+            .store(in: &cancellables)
+
+        searchBar.textField
+            .textDidEndEditing
```
```suggestion
        searchBar
            .textField
            .textDidEndEditing
```

barbarakos committed:
PR changes
